File: 151004/Eremeichik/rw/src/main/java/com/example/rw/service/db_operations/interfaces/MessageService.java
package com.example.rw.service.db_operations.interfaces;

import com.example.rw.exception.model.not_found.EntityNotFoundException;
import com.example.rw.kafkacl.response.KafkaResponse;
import com.example.rw.model.dto.message.MessageRequestTo;
import com.example.rw.model.entity.implementations.Message;
import com.example.rw.model.entity.implementations.News;
import com.fasterxml.jackson.core.JsonProcessingException;

import java.util.List;

public interface MessageService {
    KafkaResponse findById(Long id) throws JsonProcessingException;

    KafkaResponse findAll() throws JsonProcessingException;

    KafkaResponse save(MessageRequestTo entity) throws JsonProcessingException;

    void deleteById(Long id) throws JsonProcessingException;

    KafkaResponse update(MessageRequestTo request) throws JsonProcessingException;
}

File: 151004/Burbouski/appLab2/src/main/kotlin/com/github/hummel/dc/lab2/service/MessageService.kt
package com.github.hummel.dc.lab2.service

import com.github.hummel.dc.lab2.dto.request.MessageRequestTo
import com.github.hummel.dc.lab2.dto.request.MessageRequestToId
import com.github.hummel.dc.lab2.dto.response.MessageResponseTo

interface MessageService {
	suspend fun getAll(): List<MessageResponseTo>

	suspend fun create(messageRequestTo: MessageRequestTo?): MessageResponseTo?

	suspend fun deleteById(id: Long): Boolean

	suspend fun getById(id: Long): MessageResponseTo?

	suspend fun update(messageRequestToId: MessageRequestToId?): MessageResponseTo?
}